### ISSUE TITLE
Repeating seek failures in `WINDOWS/SYSTEM/DVA.386` when reinstalling WinG

### ISSUE THREAD

**Issue by interloper98** [dosbox-staging/dosbox-staging#3778]
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

0.82-alpha (main)

### What Operating System are you using?

Linux ARM (Raspberry Pi)

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

When re-installing the WinG package in Window 3.11, I get repeating warnings in the log:

```text
FS: File seek failed for './WINDOWS/SYSTEM/DVA.386'
FS: File seek failed for './WINDOWS/SYSTEM/DVA.386'
FS: File seek failed for './WINDOWS/SYSTEM/DVA.386'
FS: File seek failed for './WINDOWS/SYSTEM/DVA.386'
FS: File seek failed for './WINDOWS/SYSTEM/DVA.386'
FS: File seek failed for './WINDOWS/SYSTEM/DVA.386'
```

This is essentially locks up Windows 3.11, however the emulator can still be Ctrl+F9 to quit.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Download and extract: [wfw-trio-170.zip](https://github.com/user-attachments/files/15891109/wfw-trio-170.zip)

2. Run Staging when inside the `wfw-trio-170/` directory

3. Windows 3.11 starts, then launch the 'rerun WinG setup' icon:

    ![wing-resetup](https://github.com/dosbox-staging/dosbox-staging/assets/165201780/da44813a-16f4-4a41-86b9-55bcb88e540d)

4. Click default/yes answers; the install should jam up with these seek errors filling Staging's log output.

---

A one-time work-around is to delete `WINDOWS/SYSTEM/DVA.386` and also delete the `device=` s reference in `WINDOWS/SYSTEM.INI`:

```ini
[386Enh]
device=C:\WINDOWS\SYSTEM\dva.386
```

### Download URL of affected game or software

https://github.com/user-attachments/files/15891109/wfw-trio-170.zip

### Your configuration

```ini
`dosbox.conf` is included in the zip.
```

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by weirddan455** [comment]
Probably a regression from https://github.com/dosbox-staging/dosbox-staging/commit/39d8414067076c0baf03c70aaf62dfde8de0eade

I was unsure if anything would hit that case or what to do if it did.  I believe the old code was seeking to the end of the file rather than the beginning so might be as simple as that.  I'm going to put another file related PR in the next couple days so I'll take a look.

**Comment by interloper98** [comment]
> I was unsure if anything would hit that case or what to do if it did so I added that warning that you're seeing. 

Right on @weirddan455. That log message did it's job; glad there's a test that hits it!

**Comment by weirddan455** [comment]
I tested and both seeking to the end of the file and throwing a DOS error work to complete the install.  The reason for failure is it's trying to `SEEK_CUR` to `-16777216` (-1.6MB in a 1.9KB file).

I was referencing MS-DOS Encyclopedia but I just saw that book was written in 1988 and the latest DOS at the time was 3.3.  In the book, it says that the seek function doesn't fail if it would cause the file pointer to be placed before the beginning of the file but it would cause "unexpected error" on a later read or write.

My guess is that later versions of DOS got smarter and threw an error in that case.  I may have to bust out the assembler and test in MS-DOS 6.22 and see what such an invalid seek actually does.
